scheduler/core/programprovider/ocs/ocsprogramprovider.py

```diff
@@ -763,109 +773,111 @@ def autocorr_lag(x):
             next_atom = False
-
             observe_class = step[OcsProgramProvider._AtomKeys.OBS_CLASS]
-            step_time = step[OcsProgramProvider._AtomKeys.TOTAL_TIME] / 1000
-
-            # Any wavelength/filter change is a new atom
-            if step_id == 0 or (step_id > 0 and wavelengths[step_id] != wavelengths[step_id - 1]):
-                next_atom = True
-                # logger.info('Atom for wavelength change')
-                # print(f'\t\t\t Atom for wavelength change')
-
-            # A change in exposure time or coadds is a new atom for science exposures
-            # print(f'\t\t\t {step[OcsProgramProvider._AtomKeys.OBSERVE_TYPE].upper()}')
-            if step[OcsProgramProvider._AtomKeys.OBSERVE_TYPE].upper() not in OcsProgramProvider._OBSERVE_TYPES:
-                if (prev >= 0 and observe_class.upper() == ObservationClass.SCIENCE.name and step_id > 0 and
-                        (exposure_times[step_id] != exposure_times[prev] or coadds[step_id] != coadds[prev])):
+            if observe_class.upper() not in [ObservationClass.ACQ.name, ObservationClass.ACQCAL.name]:
+                step_use += 1
+                step_time = step[OcsProgramProvider._AtomKeys.TOTAL_TIME] / 1000
+
+                # Any wavelength/filter change is a new atom
+                if step_use == 0 or (step_use > 0 and wavelengths[step_use] != wavelengths[step_use - 1]):
                     next_atom = True
-                    # logger.info('Atom for exposure time change')
-                    # print(f'\t\t\t Atom for exposure time change')
-
-                # Offsets - a new offset pattern is a new atom
-                if offset_lag != 0 or not exp_time_groups:
-                    # For NIR imaging, need to have at least two offset positions if no repeating pattern
-                    # New atom after every 2nd offset (noffsets is odd)
-                    if (mode is ObservationMode.IMAGING and offset_lag == 0 and
-                            all(w > 1.0 for w in wavelengths if w is not None)):
-                        if step_id == 0:
-                            n_offsets += 1
-                        else:
-                            if p_offsets[step_id] != p_offsets[prev] or q_offsets[step_id] != q_offsets[prev]:
+                    # logger.info('Atom for wavelength change')
+                    # print(f'\t\t\t Atom for wavelength change')
+
+                # A change in exposure time or coadds is a new atom for science exposures
+                # print(f'\t\t\t {step[OcsProgramProvider._AtomKeys.OBSERVE_TYPE].upper()}')
+                if step[OcsProgramProvider._AtomKeys.OBSERVE_TYPE].upper() not in OcsProgramProvider._OBSERVE_TYPES:
+                    if (prev >= 0 and observe_class.upper() == ObservationClass.SCIENCE.name and step_use > 0 and
+                            (exposure_times[step_use] != exposure_times[prev] or coadds[step_use] != coadds[prev])):
```

Comment:
The logic here is rather confusing: I think it merits a comment for maintainability.